extensions/integration-tests/src/testConfig.ts

```diff
@@ -100,24 +125,19 @@ function getEnumMappingEntry(mapping: any, enumValue: any): INameDisplayNamePair
 	}
 }
 
-export async function getDefaultTestingServer(): Promise<TestServerProfile> {
-	let servers = await getTestingServers();
-	return servers[0];
-}
-
 export async function getAzureServer(): Promise<TestServerProfile> {
 	let servers = await getTestingServers();
-	return servers.filter(s => s.version === '2012')[0];
+	return servers.filter(s => s.engineType === EngineType.Azure)[0];
 }
 
 export async function getStandaloneServer(): Promise<TestServerProfile> {
 	let servers = await getTestingServers();
-	return servers.filter(s => s.version === '2017')[0];
+	return servers.filter(s => s.version === '2017' && s.engineType === EngineType.Standalone)[0];
```

Comment:
>s => s.version === '2017' [](start = 23, length = 25)

I'd just get rid of the version completely IMO. It's not useful anywhere currently and is just an annoyance we'll have to think about if we ever change the backend servers.

Reply:
i'd leave it this way, we might add different versions of test servers.

Reply:
And if we do we can add it in then and make it more complete - but right now my opinion is that leaving it in is just confusing and has the potential for testing errors if people (wrongly) assume that we'll be ensuring the server is actually 2017/2019/whatever. Just because the servers happen to be now doesn't mean tomorrow we might decide to update the on-prem server to 2019. (Which honestly we should do - testing against the latest version is always a good idea).